openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -30,48 +30,263 @@ export function closePop(){
     });
 }
 
-export default function init(){
-    var $searchResults = $('header#header-bar .search-component ul.search-results');
-    var $searchInput = $('header#header-bar .search-component .search-bar-input input[type="text"]');
-    var cover_url = function(id) {
-        return `//covers.openlibrary.org/b/id/${id}-S.jpg`
-    };
-    // stores the state of the search result for resizing window
-    var instantSearchResultState = false;
-    var searchModes, searchModeDefault, defaultFacet, searchFacets, composeSearchUrl, marshalBookSearchQuery, renderInstantSearchResults, setFacet, setMode, setSearchMode, options, q, parts, enteredSearchMinimized, searchExpansionActivated, toggleSearchbar, renderInstantSearchResult, val, facet_value;
-
-    // searches should be cancelled if you click anywhere in the page
-    $('body').on('click', function () {
-        instantSearchResultState = false;
-        $searchResults.empty();
-    });
-    // but clicking search input should not empty search results.
-    $searchInput.on('click', false);
+const SEARCH_MODES = ['everything', 'ebooks', 'printdisabled'];
+const SEARCH_MODE_DEFAULT = 'ebooks';
+const SEARCH_FACETS = {
+    'title': 'books',
+    'author': 'authors',
+    'lists': 'lists',
+    'subject': 'subjects',
+    'all': 'all',
+    'advanced': 'advancedsearch',
+    'text': 'inside'
+};
+const DEFAULT_SEARCH_FACET = 'all';
+const RENDER_INSTANT_SEARCH_RESULT = {
+    books(work) {
+        const author_name = work.author_name ? work.author_name[0] : '';
+        $('header#header-bar .search-component ul.search-results').append(
+            `<li class="instant-result">
+                <a href="${work.key}">
+                    <img src="//covers.openlibrary.org/b/id/${work.cover_i}-S.jpg"/>
+                    <span class="book-desc">
+                        <div class="book-title">${work.title}</div> by <span class="book-author">${author_name}</span>
+                    </span>
+                </a>
+            </li>`
+        );
+    },
+    authors(author) {
+        // Todo: default author img to: https://dev.openlibrary.org/images/icons/avatar_author-lg.png
+        $('header#header-bar .search-component ul.search-results').append(
+            `<li>
```

Comment:
I thought your idea to create html-test-data.js in tests was useful.
Let's do the same here
e.g. `search/components.js`

export const searchResultItem = `....`

Reply:
I'm wary to do that here? In the test files, we mostly just have plain strings, not templates (i.e. functions), which we've copy/pasted from the website. Those are SUPER likely to break because if the templates change, that's why I think it's important to "fence them off" until we find a better solution. Templates here are also nonideal, but they interact with the code more and aren't as brittle as the string in the tests. I think there would be a loss of readability if they're separated here. I'm ok leaving them until we come up with our solution for how to sync server/client templates. So WONTFIX